be/src/olap/like_column_predicate.cpp

```diff
@@ -23,74 +23,131 @@
 
 namespace doris {
 
-LikeColumnPredicate::LikeColumnPredicate(bool opposite, uint32_t column_id,
-                                         doris_udf::FunctionContext* fn_ctx,
-                                         doris_udf::StringVal val)
+template <>
+LikeColumnPredicate<true>::LikeColumnPredicate(bool opposite, uint32_t column_id,
+                                               doris_udf::FunctionContext* fn_ctx,
+                                               doris_udf::StringVal val)
+        : ColumnPredicate(column_id, opposite),
+          _fn_ctx(fn_ctx),
+          pattern(reinterpret_cast<char*>(val.ptr), val.len) {
+    _state = reinterpret_cast<StateType*>(
+            _fn_ctx->get_function_state(doris_udf::FunctionContext::THREAD_LOCAL));
+    _state->search_state.clone(_like_state);
```

Comment:
So here clone the state because of the Hyperscan? Not sure the overhead of clone is high or not.

Reply:
Yes, every concurrent caller of the Hyperscan API should use a separate scratch. Here each LikeColumnPredicate only clones it once, so I think the overload is acceptable.

Reply:
OK